torch/ao/quantization/fx/_model_report/model_report_visualizer.py

```diff
@@ -102,135 +112,256 @@ def get_all_unique_feature_names(self, plottable_features_only: bool = True) ->
+                    # we iterate in same order of added features
+
+                    if feature in filtered_data[module_fqn]:
+                        # add value if applicable to module
+                        feature_val = filtered_data[module_fqn][feature]
+                    else:
+                        # add that it is not applicable
+                        feature_val = "Not Applicable"
+
+                    # if it's a tensor we want to extract val
+                    if type(feature_val) is torch.Tensor:
+                        feature_val = feature_val.item()
+
+                    # we add to our list of values
+                    tensor_table_row.append(feature_val)
+
+                tensor_table.append(tensor_table_row)
+
+        # add row of headers of we actually have something, otherwise just empty
+        if len(tensor_table) == 0:
+            tensor_table.append([])  # just an empty list to signify no data
+        else:
+            tensor_headers: List[str] = ["idx", "layer_fqn"] + tensor_features_list
+            tensor_table.insert(0, tensor_headers)
+
+        # now we compose the table for the channel information table
+        channel_table: List[List[Any]] = []
+
+        # counter to keep track of number of entries in
+        channel_table_entry_counter: int = 1
+
+        if len(channel_features_list) > 0:
+            # now we add all channel data
+            for index, module_fqn in enumerate(filtered_data):
+                # we iterate over all channels
+                for channel in range(num_channels):
+                    # we make a new row for the channel
+                    new_channel_row = [channel_table_entry_counter, module_fqn, channel]
+                    for feature in channel_features_list:
+                        if feature in filtered_data[module_fqn]:
+                            # add value if applicable to module
+                            feature_val = filtered_data[module_fqn][feature][channel]
+                        else:
+                            # add that it is not applicable
+                            feature_val = "Not Applicable"
+
+                        # if it's a tensor we want to extract val
+                        if type(feature_val) is torch.Tensor:
+                            feature_val = feature_val.item()
+
+                        # add value to channel specific row
+                        new_channel_row.append(feature_val)
+
+                    # add to table and increment row index counter
+                    channel_table.append(new_channel_row)
+                    channel_table_entry_counter += 1
+
+        # add row of headers of we actually have something, otherwise just empty
+        if len(channel_table) == 0:
+            channel_table.append([])  # just an empty list to signify no data
+        else:
+            channel_headers: List[str] = ["idx", "layer_fqn", "channel"] + channel_features_list
```

Comment:
should headers be a part of the table as well? this looks like a metadata for the table. maybe we should define a object to represent table?

Reply:
i don't feel like the table is enough of a thing that it warrants its own class, it's only attributes are its headers and the table, however if you believe it should get one we can make the change